src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs

```diff
@@ -144,10 +144,19 @@ private List<ProductInfoHeaderValue> DefaultUserAgentInfoList
                 if(_defaultUserAgentInfoList == null)
                 {
                     _defaultUserAgentInfoList = new List<ProductInfoHeaderValue>();
-                    _defaultUserAgentInfoList.Add(new ProductInfoHeaderValue(FXVERSION, FrameworkVersion));
+                    if(!String.IsNullOrWhiteSpace(FrameworkVersion))
+                    {
+                        _defaultUserAgentInfoList.Add(new ProductInfoHeaderValue(FXVERSION, FrameworkVersion));
+                    }
 #if FullNetFx
-                    _defaultUserAgentInfoList.Add(new ProductInfoHeaderValue(OSNAME, OsName));
-                    _defaultUserAgentInfoList.Add(new ProductInfoHeaderValue(OSVERSION, OsVersion));
+                    if(!String.IsNullOrWhiteSpace(OsName))
+                    {
+                        _defaultUserAgentInfoList.Add(new ProductInfoHeaderValue(OSNAME, OsName));
+                    }
+                    if(!String.IsNullOrWhiteSpace(OsVersion))
+                    {
+                        _defaultUserAgentInfoList.Add(new ProductInfoHeaderValue(OSVERSION, OsVersion));
+                    }
```

Comment:
@dsgouda we also need to cover/handle ClientVersion. What if ClientVersion also has null/empty value.

Reply:
Good catch, addressing.